src/Admin/Polyfills.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Handles backward compatibility of assets for older versions of WP.
+ *
+ * @since 2.0
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Scripts;
+use WP_Styles;
+
+/**
+ * Registers assets that may not be available in the current site's version of core.
+ *
+ * @since 2.0
+ */
+final class Polyfills implements Service, Registerable {
+
+	/**
+	 * Runs on instantiation.
+	 */
+	public function register() {
+		add_action( 'wp_default_scripts', [ $this, 'register_shimmed_scripts' ], 11 ); // Runs immediately after core sets up default scripts.
+		add_action( 'wp_default_styles', [ $this, 'register_shimmed_styles' ], 11 ); // Runs immediately after core sets up default styles.
+	}
+
+	/**
+	 * Registers scripts not guaranteed to be available in core.
+	 *
+	 * @param WP_Scripts $wp_scripts The WP_Scripts instance for the current page.
+	 */
+	public function register_shimmed_scripts( $wp_scripts ) {
+		if ( ! isset( $wp_scripts->registered['lodash'] ) ) {
+			$wp_scripts->add(
+				'lodash',
+				amp_get_asset_url( 'js/vendor/lodash.js' ),
+				[],
+				'4.17.19',
+				true
+			);
+
+			$wp_scripts->add_inline_script( 'lodash', 'window.lodash = _.noConflict();' );
+		}
+
+		/*
+		* Polyfill dependencies that are registered in Gutenberg and WordPress 5.0.
+		* Note that Gutenberg will override these at wp_enqueue_scripts if it is active.
+		*/
+		$handles = [ 'wp-i18n', 'wp-dom-ready', 'wp-polyfill', 'wp-url' ];
+		foreach ( $handles as $handle ) {
+			if ( ! isset( $wp_scripts->registered[ $handle ] ) ) {
+				$asset_file   = AMP__DIR__ . '/assets/js/' . $handle . '.asset.php';
```

Comment:
We're defaulting to curly brace substitution in PHP:
```suggestion
				$asset_file   = AMP__DIR__ . "/assets/js/{$handle}.asset.php";
```